core/domain/prod_validators.py

```diff
@@ -4719,10 +4721,30 @@ def _validate_sender_and_recipient_ids(
                         'recipient with id %s' % (
                             item.id, recipient_user_ids[index]))
 
+    @classmethod
+    def _validate_old_models_are_deleted(cls, item):
+        """Validate that there are no models that were last updated more than
+        twelve weeks ago, these models should be deleted.
+
+        Args:
+            item: UserQueryModel. UserQueryModel to validate.
+        """
+        date_four_weeks_ago = (
+            datetime.datetime.utcnow() - UserQueryModelValidator.FOUR_WEEKS)
+        if item.last_updated < date_four_weeks_ago:
+            cls._add_error(
+                'entity %s' % base_model_validators.ERROR_CATEGORY_STALE_CHECK,
+                'Entity id %s: Model older than 4 weeks' % item.id
+            )
+
```

Comment:
Do we need to validate that the models are marked delete after 4 weeks?

Reply:
We already do this for the `UserQueryModel`.

Reply:
I only saw the validation for "no model is older than 12 weeks.". I may have missed that can you please share the link?

Reply:
Explained in a different comment.